tests/install_target_test.go

```diff
@@ -0,0 +1,151 @@
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("kairos install test different targets", Label("install-test-target"), func() {
+
+	var vm VM
+	var label string
+	var diskUUID uuid.UUID
+	BeforeEach(func() {
+		label = "TESTDISK"
+		diskUUID = uuid.New()
+		_, vm = startVM()
+		vm.EventuallyConnects(1200)
+		// Format the disk so it gets an uuid and label
+		_, err := vm.Sudo(fmt.Sprintf("mkfs.ext4 -L %s -U %s /dev/vda", label, diskUUID.String()))
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+		Expect(vm.Destroy(nil)).ToNot(HaveOccurred())
+	})
+
+	Context("Selects the disk by uuid/label", func() {
+		It("Selects the correct disk if using uuid for target", func() {
+			expectSecureBootEnabled(vm)
+
+			err := vm.Scp("assets/config.yaml", "/tmp/config.yaml", "0770")
+			Expect(err).ToNot(HaveOccurred())
+
+			var out string
+			// Test that install works
+			By("installing kairos", func() {
+				installCmd := fmt.Sprintf("kairos-agent --debug manual-install --device /dev/disk/by-uuid/%s /tmp/config.yaml", diskUUID.String())
+				By(fmt.Sprintf("Running %s", installCmd))
+				out, err = vm.Sudo(installCmd)
+				Expect(err).ToNot(HaveOccurred(), out)
+				Expect(out).Should(ContainSubstring("Running after-install hook"))
+				vm.Sudo("sync")
+			})
+
+			By("waiting for VM to reboot", func() {
+				vm.Reboot()
+				vm.EventuallyConnects(1200)
+			})
+
+			By("checking that vm has rebooted to 'active'", func() {
+				Eventually(func() string {
+					out, _ := vm.Sudo("kairos-agent state boot")
+					return out
+				}, 40*time.Minute, 10*time.Second).Should(
+					Or(
+						ContainSubstring("active_boot"),
+					))
+			})
+
+			By("checking corresponding state", func() {
+				out, err := vm.Sudo("kairos-agent state")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(out).To(ContainSubstring("boot: active_boot"))
+				currentVersion, err := vm.Sudo(getVersionCmd)
+				Expect(err).ToNot(HaveOccurred(), currentVersion)
+
+				stateAssertVM(vm, "oem.mounted", "true")
+				stateAssertVM(vm, "oem.found", "true")
+				stateAssertVM(vm, "persistent.mounted", "true")
+				stateAssertVM(vm, "state.mounted", "true")
+				stateAssertVM(vm, "oem.type", "ext4")
+				stateAssertVM(vm, "persistent.type", "ext4")
+				stateAssertVM(vm, "state.type", "ext4")
+				stateAssertVM(vm, "oem.mount_point", "/oem")
+				stateAssertVM(vm, "persistent.mount_point", "/usr/local")
+				stateAssertVM(vm, "persistent.name", "/dev/vda")
+				stateAssertVM(vm, "state.mount_point", "/run/initramfs/cos-state")
+				stateAssertVM(vm, "oem.read_only", "false")
+				stateAssertVM(vm, "persistent.read_only", "false")
+				stateAssertVM(vm, "state.read_only", "true")
+				stateAssertVM(vm, "kairos.version", strings.ReplaceAll(strings.ReplaceAll(currentVersion, "\r", ""), "\n", ""))
+				stateContains(vm, "system.os.name", "alpine", "opensuse", "ubuntu", "debian")
+				stateContains(vm, "kairos.flavor", "alpine", "opensuse", "ubuntu", "debian")
+			})
+		})
```

Comment:
I don't see any check that the correct disk was actually used. We only check that the command didn't fail and that installation occurred. I guess, since we only have one disk, it was indeed selected but would be better if we had 2 disks and we made sure the correct one was selected, to avoid for example the flag being completely ignored and the default disk (bigger one?) gets selected instead.

But it can be tricky: https://github.com/kairos-io/kairos/blob/15f342f8cb3bcc8c00aba06dedcc9bc5362b7aea/tests/custom_partitioning_test.go#L68-L76

I don't have a strong opinion, I'll let you decide.

Reply:
Yes indeed, that's a good point. I am letting it work as is because if you pass a wrong or nonexistent disk, the install fails but I'll have a look at peg to see if we can pass multiple disks to the VM, that would make it more robust

Reply:
https://github.com/spectrocloud/peg/commit/c12a7362c36c1c945ccc1c4e98e5ffee0808b7cb

Reply:
I introduced it for this exact reason. It's the boot order that is hard to do in qemu I think (see the link above).

Reply:
but boot order should not matter no? It will try first all the disks and then livecd? Or it just tries the first disk only?

Reply:
You are rigth, it doesnt try to boot from the second one somehow :sob:

Reply:
Maybe we can add a default bootindex to all disks in qemu in peg, as in: first check disks (/dev/vda, /dev/vdb, etc) then cdrom.
This should allow us to try to target the second disk and see if it worked (since vda will be skipped as not being bootable).